ygen/codegen.go

```diff
@@ -885,11 +894,29 @@ type mappedYANGDefinitions struct {
 // It returns a mappedYANGDefinitions struct populated with the directory, enum
 // entries in the input schemas as well as the calculated schema tree.
 func mappedDefinitions(yangFiles, includePaths []string, cfg *GeneratorConfig) (*mappedYANGDefinitions, util.Errors) {
-	modules, errs := processModules(yangFiles, includePaths, cfg.ParseOptions.YANGParseOptions)
+	moduleEntries, errs := processModules(yangFiles, includePaths, cfg.ParseOptions.YANGParseOptions)
 	if errs != nil {
 		return nil, errs
 	}
 
+	// Exclude any modules that do not appear in the yangFiles list.
+	if cfg.ExcludeSearchPathModules {
+		inFiles := map[string]bool{}
+		for _, f := range yangFiles {
+			inFiles[f] = true
+		}
+		for _, module := range moduleEntries {
+			// TODO(wenbli): implement a better way to get the source file for a yang module.
+			moduleSrcFile := strings.Split(module.Node.Statement().Location(), ":")[0]
+			if moduleSrcFile == "" {
+				return nil, append(errs, fmt.Errorf("module %q source file was empty", module.Name))
+			}
+			if !inFiles[moduleSrcFile] {
+				cfg.ParseOptions.ExcludeModules = append(cfg.ParseOptions.ExcludeModules, module.Name)
```

Comment:
Modify excluded instead of the input options.

Reply:
changed findMappableEntities to take map as well